src/build-config.ts

```diff
@@ -397,22 +447,52 @@ async function buildConfig(
     tsOptions,
     dts,
   )
-  const outputExports = getExportConditionDist(pkg, exportCondition, cwd)
+  const outputExports: Array<string | { format: 'cjs' | 'esm'; file: string }> =
+    dts
+      ? getExportTypeDist(exportCondition, cwd)
+      : getExportConditionDist(pkg, exportCondition, cwd)
+
+  if (pkg.bin) {
+    const isSinglePathBin = typeof pkg.bin === 'string'
+    const binDistPaths = isSinglePathBin
+      ? [pkg.bin as string]
+      : Object.values(pkg.bin)
+
+    for (const binDistPath of binDistPaths) {
+      const ext = extname(binDistPath).slice(1) as 'js' | 'cjs' | 'mjs'
+
+      if (dts) {
+        const dtsExt = dtsExtentions[ext]
+        const filename = filenameWithoutExtension(binDistPath)
+        if (!filename) continue
+
+        const binTypeFile = `${filename}${dtsExt}`
+        outputExports.push(binTypeFile)
+      } else {
+        // ESM by default, CJS if the file extension is .cjs
+        const isCJS = ext === 'cjs'
```

Comment:
Could there be a better way to know the bin files' format other than the output extension?

Reply:
It's based on package.json type, If it's `type: module`, then `.js` are treated as ESM. and only `.cjs` will be treated as cjs.
Otherwise, the .js files will be treated as cjs, and only mjs will be ESM